db/flush_scheduler.cc

```diff
@@ -87,4 +87,50 @@ void FlushScheduler::Clear() {
   assert(head_.load(std::memory_order_relaxed) == nullptr);
 }
 
+void TrimHistoryScheduler::ScheduleWork(ColumnFamilyData* cfd) {
+  std::lock_guard<std::mutex> lock(checking_mutex_);
+  assert(checking_set_.count(cfd) == 0);
+  checking_set_.insert(cfd);
+  cfd->Ref();
+  cfds.push_back(cfd);
+}
+
+ColumnFamilyData* TrimHistoryScheduler::TakeNextColumnFamily() {
+  std::lock_guard<std::mutex> lock(checking_mutex_);
+  while (true) {
+    if (cfds.empty()) {
+      return nullptr;
+    }
+    ColumnFamilyData* cfd = cfds.back();
+    cfds.pop_back();
+    auto iter = checking_set_.find(cfd);
+    assert(iter != checking_set_.end());
+    checking_set_.erase(iter);
```

Comment:
Also, in release build, things are inserted, found and erased in checking_set_ with no validation. It's wasted work.
Either guard all the checking set related code within debug builds, or just remove all of them. TrimHistoryScheduler doesn't do lock-free algorithm so this validation is less needed than in FlushScheduler. If we really want to validate the correctness, a better approach is to write a new unit test for it.

Reply:
removed the checking set